addons/nuxeo-platform-spreadsheet/nuxeo-platform-spreadsheet-web/src/main/js/app/index.jsp

```diff
@@ -74,7 +74,7 @@
   <script>
     var nuxeo = nuxeo || {};
     nuxeo.spreadsheet = {
-      language: '<%= selectedLanguage %>',
+      language: navigator.language || '<%= selectedLanguage %>',
```

Comment:
True, sounds sensible.
We can change the approach to have cache first, then navigator language and, as fallback, english. It may tackle most of the scenarios.